### PR TITLE
Homework 5. Boris Tsarikov ( Updated )

### PR DESCRIPTION
**Pull request by B0RlS**
The main functionality is there, an additional implementation has been added. There may be problems with business logic, waiting for comments on the project to fix (if necessary)
^_^

### CODE REVIEW

**Comment by B0RlS** [comment]
There is a problem in initialize with super, I will ask about it for classes
class Student <Human
   attr_accessor: mentors,: homework,: repository

   def initialize (name :, surname :, id :, homework :, repository :)
     @homework = homework
     @repository = repository
     @mentors = []
     super
   end

Mistake
home / boris / EPAM / Work / Homework / homeworks / HW_05 / Boris_Tsarikov / human.rb: 5: in `initialize ': unknown keywords: homework, repository (ArgumentError)

I had to put the superclass fields as an heir, the chopping down is scolded by the number of arguments in initialize> 5

File: HW_05/Boris_Tsarikov/entity/homework.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Top level class describing a homework
+class Homework
```
Comment (labs-ruby) [inline HW_05/Boris_Tsarikov/entity/homework.rb]: Eventually, what public interface does the homework provide?

File: HW_05/Boris_Tsarikov/entity/student.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
+require 'json'
+# Heir to the class Human, describes the student and his work
+class Student < Human
+  attr_accessor :mentors, :homework, :repository, :mentors_list
+
+  def initialize(fullname:, id:, homework:, repository:, mentors_list:)
+    @homework = homework
+    @repository = repository
+    @mentors = []
+    @mentors_list = mentors_list
```
Comment (labs-ruby) [inline HW_05/Boris_Tsarikov/entity/student.rb]: Why do you pass `mentor_list` to student's constructor?

File: HW_05/Boris_Tsarikov/entity/mentor.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
+require 'json'
+# Heir to the class Human, describes the mentor and his work
+class Mentor < Human
+  attr_accessor :students, :journal, :notifications, :task, :repository
+
+  def initialize(fullname:, id:, repository:, journal:, task:)
+    @journal = journal
+    @task = task
```
[the remark this answers is not in the document]
Comment (B0RlS) [reply HW_05/Boris_Tsarikov/entity/mentor.rb]: > Why task is passed to mentor's constructor?

An instance of the Task class is transmitted, which will define the mentor and place it on the repository. Ie the mentor should know what task he will give out to students, prepare it and send

File: HW_05/Boris_Tsarikov/entity/student.rb (hunk 2 above)
Comment (B0RlS) [reply HW_05/Boris_Tsarikov/entity/student.rb]: Will fix

File: HW_05/Boris_Tsarikov/entity/homework.rb (hunk 1 above)
Comment (B0RlS) [reply HW_05/Boris_Tsarikov/entity/homework.rb]: Will fix

File: HW_05/Boris_Tsarikov/entity/journal.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# Class for storing student information
+class Journal
+  attr_accessor :student_journal, :notifications
+
+  def initialize
+    @student_journal = 'resources/journal.json'
+    @notifications = 'resources/notifications.json'
```
Comment (labs-ruby) [inline HW_05/Boris_Tsarikov/entity/journal.rb]: I suggest to not hardcode the JSON files, but pass them as an arguments
```ruby
def initialize(journal_path: 'resources/journal.json', notifications_path: 'resources/notifications.json')
   # ...
end

end
```

File: HW_05/Boris_Tsarikov/runner/app.rb [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+# Create a task object
+task_for_students = Task.new(task_name: 'none', task: 'none')
+
+# Create a journal object for mentors
+mentors_journal = Journal.new
+
+# Create a mentor object
+first_mentor = Mentor.new(fullname: 'Alexander Shagov',
+                          id: 1, repository: homework_hub,
+                          journal: mentors_journal, task: task_for_students)
```
Comment (labs-ruby) [inline HW_05/Boris_Tsarikov/runner/app.rb]: I really don't like the idea that you pass a `task` to the mentor's constructor, could you please think about passing it as an arguments to the methods depending on the task?

For now, it's quite confusing to have coupling between task and mentor 1-1

**Pull request merged**